mockgcp/mockbigquerydatatransfer/datatransfer.go

```diff
@@ -106,14 +106,25 @@ func (s *dataTransferService) CreateTransferConfig(ctx context.Context, req *pb.
 	if obj.ScheduleOptionsV2 == nil {
 		obj.NextRunTime = timestamppb.New(now)
 	}
+	// Update ScheduleOptionV2 for time based schedule
+	if obj.Schedule != "" {
+		obj.ScheduleOptionsV2 = &pb.ScheduleOptionsV2{
+			Schedule: &pb.ScheduleOptionsV2_TimeBasedSchedule{TimeBasedSchedule: &pb.TimeBasedSchedule{Schedule: obj.Schedule}},
+		}
+	}
 	email := "user@google.com"
 	obj.OwnerInfo = &pb.UserInfo{
 		Email: &email,
 	}
-	obj.State = pb.TransferState_PENDING
+	// field was not populated in realGCP log, commented out temporarily
+	//obj.State = pb.TransferState_PENDING
 	obj.UpdateTime = timestamppb.New(now)
 	obj.UserId = int64(123)
 
+	if obj.EncryptionConfiguration != nil && obj.EncryptionConfiguration.KmsKeyName != nil {
+		obj.Params.Fields["destination_table_kms_key"] = structpb.NewStringValue(obj.EncryptionConfiguration.KmsKeyName.Value)
```

Comment:
Just curious, is `obj.Params.Fields` always an established map? I'm a bit concerned about the potential nil pointer issue here.